_posts/2020-06-21-one-stop-guide-to-lombok.md

```diff
@@ -0,0 +1,337 @@
+The most useful Lombok feature is `@NonNull`. In that case Lombok adds null-checks in you code. It will look like:  
+```java
+if (name == null) {
+  throw new NullPointerException("name is marked non-null but is null");
+}
+```
+
+## @Slf4j annotation
+
+One of the usefull Lombok features is `@Slf4j`. Lombok will generate static Slf4j Logger `log` property in your class which you can use to log something with choosen severity.  
+
+```java
+log.info("myMethod(): started");
+log.debug("myMethod(): doing some calculations");
+log.info("myMethod(): finished");
+```
+
+## @Builder annotation
+
+If use often use Builder pattern you will like the next Lombok feature. It is `@Builder`. You can annotate your class with this feature and Lombok will generate all neccessery code.  
+
+Lombok annotated code
+
+```java
+@Builder
+public class Person {
+    private String name;
+    private Integer age;
+}
+
+```
+
+Java generated code
+
+```java
+public class Person {
+    private String name;
+    private Integer age;
+
+    Person(final String name, final Integer age) {
+        this.name = name;
+        this.age = age;
+    }
+
+    public static Person.PersonBuilder builder() {
+        return new Person.PersonBuilder();
+    }
+
+    public static class PersonBuilder {
+        private String name;
+        private Integer age;
+
+        PersonBuilder() {
+        }
+
+        public Person.PersonBuilder name(final String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Person.PersonBuilder age(final Integer age) {
+            this.age = age;
+            return this;
+        }
+
+        public Person build() {
+            return new Person(this.name, this.age);
+        }
+
+        public String toString() {
+            return "Person.PersonBuilder(name=" + this.name + ", age=" + this.age + ")";
+        }
+    }
+}
+```
+
+## @Data annotation
+This annotation is used the most frequently amongst all other Lombok annotations. It combines the functionality of the next Lombok annotations: `@ToString`, `@EqualsAndHashCode`, `@Getter`, `@Setter`,
+`@RequiredArgsConstructor`
+
+# How to add the support of Lombok in your project
+
+## Dependencies to add for Maven
+
+To set up lombok with Maven, you have to specify that the lombok dependency is required to compile your source code, but does not need to be present when running/testing/jarring/otherwise deploying your code. Generally this is called a 'provided' dependency.  
+Lombok is available in maven central, so you can easily add it in `dependencies` section of your `pom.xml` file.
+
+```java
+<dependencies>
+  <dependency>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.18.12</version>
+    <scope>provided</scope>
+  </dependency>
+</dependencies>
+```
+
+*N.B.* ~when adding lombok annotations make sure that you have imported them from correct package. This is the correct import for annotation `@Getter`  
```

Comment:
Don't use abbreviations like "N.B", instead use normal text to note things. Or introduce it with "Note that, ..."

Reply:
Why do I need to take care to take the annotation from the right package?